spec/Section 6 -- Execution.md

```diff
@@ -322,39 +312,290 @@ Unsubscribe(responseStream):
+GetPendingEntry(pendingResults):
+
+- Initialize {pending} to an empty list.
+- For each {pendingResult} of {pendingResult}:
+  - Let {id} be a unique identifier for {pendingResult}.
+  - Let {path} and {label} be the corresponding entries on {pendingResult}.
+  - Let {pendingEntry} be an unordered map containing {id}, {path}, and {label}.
+  - Append {pendingEntry} to {pending}.
+- Return {pending}.
+
+GetIncrementalResult(graph, incremental, completed, pending):
+
+- Let {hasNext} be {false} if {graph} is empty, otherwise, {true}.
+- Let {incrementalResult} be an unordered map containing {hasNext}.
+- If {incremental} is not empty:
+  - Set the corresponding entry on {incrementalResult} to {incremental}.
+- If {completed} is not empty:
+  - Set the corresponding entry on {incrementalResult} to {completed}.
+- If {pending} is not empty:
+  - Set the corresponding entry on {incrementalResult} to {pending}.
+- Return {incrementalResult}.
+
+GetIncrementalEntry(incrementalDataRecord, graph):
+
+- Let {deferredFragments} be the Deferred Fragments incrementally completed by
+  {incrementalDataRecord} at {path}.
+- Let {result} be the result of {incrementalDataRecord}.
+- Let {data} and {errors} be the corresponding entries on {result}.
+- Let {releasedDeferredFragments} be the members of {deferredFragments} that are
+  root nodes in {graph}.
+- Let {bestDeferredFragment} be the member of {releasedDeferredFragments} with
+  the shortest {path} entry.
+- Let {subPath} be the portion of {path} not contained by the {path} entry of
+  {bestDeferredFragment}.
+- Let {id} be the unique identifier for {bestDeferredFragment}.
+- Return an unordered map containing {id}, {subPath}, {data}, and {errors}.
+
+GetCompletedEntry(pendingResult, errors):
+
+- Let {id} be the unique identifier for {pendingResult}.
+- Let {completedEntry} be an unordered map containing {id}.
+- If {errors} is not empty, set the corresponding entry on {completedEntry} to
+  {errors}.
+- Return {completedEntry}.
+
+### Batching Incremental Results
+
+BatchIncrementalResults(incrementalResults):
+
+- Return a new stream {batchedIncrementalResults} which yields events as
+  follows:
+- While {incrementalResults} is not closed:
+  - Let {availableIncrementalResults} be a list of one or more Incremental
+    Results available on {incrementalResults}.
+  - Let {batchedIncrementalResult} be an unordered map created by merging the
+    items in {availableIncrementalResults} into a single unordered map,
+    concatenating list entries as necessary, and setting {hasNext} to the value
+    of {hasNext} on the final item in the list.
+  - Yield {batchedIncrementalResult}.
+
+## Executing an Execution Plan
+
+To execute a execution plan, the object value being evaluated and the object
+type need to be known, as well as whether the non-deferred grouped field set
+must be executed serially, or may be executed in parallel.
+
+ExecuteExecutionPlan(newDeferUsages, executionPlan, objectType, objectValue,
```

Comment:
this function is equivalent to `executeExecutionGroup` in graphql-js?

Reply:
It needs its own function, see https://github.com/graphql/graphql-js/pull/4178